tests/unit/peapods/runtimes/gateway/grpc/test_grpc_gateway_runtime.py

```diff
@@ -134,222 +135,252 @@ async def task_wrapper():
 
 
 def test_grpc_gateway_runtime_handle_messages_linear(linear_graph_dict, monkeypatch):
-    monkeypatch.setattr(
-        networking.GrpcConnectionPool,
-        'send_messages_once',
-        DummyMockConnectionPool.send_messages_once,
-    )
-    port_in = random_port()
-
-    with GRPCGatewayRuntime(
-        set_gateway_parser().parse_args(
-            [
-                '--port-expose',
-                f'{port_in}',
-                '--graph-description',
-                f'{json.dumps(linear_graph_dict)}',
-                '--pods-addresses',
-                '{}',
-            ]
+    def process_wrapper():
+        monkeypatch.setattr(
+            networking.GrpcConnectionPool,
+            'send_messages_once',
+            DummyMockConnectionPool.send_messages_once,
         )
-    ) as runtime:
+        port_in = random_port()
 
-        async def _test():
-            responses = []
-            req = request_generator(
-                '/', DocumentArray([Document(text='client0-Request')])
+        with GRPCGatewayRuntime(
+            set_gateway_parser().parse_args(
+                [
+                    '--port-expose',
+                    f'{port_in}',
+                    '--graph-description',
+                    f'{json.dumps(linear_graph_dict)}',
+                    '--pods-addresses',
+                    '{}',
+                ]
             )
-            async for resp in runtime.streamer.Call(request_iterator=req):
-                responses.append(resp)
-            return responses
+        ) as runtime:
 
-        responses = asyncio.run(_test())
-    assert len(responses) > 0
-    assert len(responses[0].docs) == 1
-    assert (
-        responses[0].docs[0].text
-        == f'client0-Request-client0-pod0-client0-pod1-client0-pod2-client0-pod3'
-    )
+            async def _test():
+                responses = []
+                req = request_generator(
+                    '/', DocumentArray([Document(text='client0-Request')])
+                )
+                async for resp in runtime.streamer.Call(request_iterator=req):
+                    responses.append(resp)
+                return responses
+
+            responses = asyncio.run(_test())
+        assert len(responses) > 0
```

Comment:
These tests need to be redefined. U need to put the `responses` in some `queue` ans assert the response from the main pytest process

Reply:
I agree that design is pretty hacky now, but it works as I am checking the exit code and any test failure will yield a non zero exit code. But it will hide the actual problem in the test.